### DISCUSSION TITLE
[New Feature] Guidelines for using inpaint in A1111

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#1143] [General]
Now the ControlNet Inpaint can directly use the A1111 inpaint path to support perfect seamless inpaint experience. It supports arbitary base model without merging and works perfectly with LoRAs and every other addons. 

This inpaint implementation is technically correct and will NOT make unwanted modifications to unmasked areas.

To use it, update your ControlNet to latest version, restart completely including your terminal, and go to A1111's img2img inpaint, open ControlNet, set preprocessor as "inpaint_global_harmonious" and use model "control_v11p_sd15_inpaint", enable it. 

You do not need to add image to ControlNet. 

This is my setting
![image](https://user-images.githubusercontent.com/19834515/235911328-42d740fe-c3dd-4602-b279-e544b4e1b084.png)

Then you can just go to A1111's inpaint and inpaint images as normal. Your images will be improved by ControlNet automatically. 

We support both inpaint "Whole picture" and "Only masked".

For example this is my setting
![image](https://user-images.githubusercontent.com/19834515/235911454-51896187-568e-4152-bae5-454c1448331e.png)

This is A1111 inpainting with seed 12345 (whole picture) with ControlNet turned OFF:
![image](https://user-images.githubusercontent.com/19834515/235911633-95ad1eb7-9568-4c2f-bd3d-13811fbc7a4b.png)

This is A1111 inpainting with seed 12345 (whole picture) with ControlNet turned ON:
![image](https://user-images.githubusercontent.com/19834515/235912181-5ddaddd8-95a1-4965-9bb3-e0732852ac57.png)

**Comment by lllyasviel** [comment]
ControlNet OFF:
![image](https://user-images.githubusercontent.com/19834515/233767168-73e39765-47e8-40ba-8293-bc6d0490aa06.png)
![image](https://user-images.githubusercontent.com/19834515/233767179-7e4d25ce-2539-4355-a1a0-28f12eababe9.png)

ControlNet ON:
![image](https://user-images.githubusercontent.com/19834515/233767217-768063c3-7717-4e9c-a63e-589c5f45cdba.png)
![image](https://user-images.githubusercontent.com/19834515/233767282-9865813d-46d9-4f9c-8d9d-2dfef41e4329.png)

**Comment by ghpkishore** [comment]
@lllyasviel any ideas on how to translate this inpainting to diffusers library. MultiControlnet with inpainting in diffusers doesn't exist as of now. Given that you have been able to implement it in A1111 extension, any suggestions or leads on how to do it for diffusers would proves really helpful.

**Comment by Rkkss** [comment]
can ip2p model work with inpaint model somehow?

**Comment by duffercn** [comment]
I am trying to use controlnet to inpaint a product photo, by which I mean I am inpainting a background for a product.
I tried this preprocessor/model pair but doesn't seems working. 
Will the model be the right one to fulfill my needs? Thank you.

**Comment by PaperOrb** [comment]
Is there a specific reason you use euler A in the examples, or is that just a coincidence?

edit: it looks like this isn't present in the readme. And another question: can you add a bit more detail about how this differs from regular inpainting models without going into too much technical detail, or maybe explain the technical stuff and then add a short dumbed down answer at the end?

**Comment by ride5k** [comment]
i'm curious about the difference between inpaint_only and inpaint_global_harmonious

**Comment by psykokwak-com** [comment]
Hello all,
Do you plan to offer the inpaint model for SDXL ?
Thanks.

**Comment by arafatx** [comment]
Can I use other inpainting model like Dreamshaper for this extension. I'm new to this

**Comment by ywaby** [comment]
any workflow for comfyui?

**Comment by maguro97** [comment]
In order to avoid unwanted modifications to unmasked areas, which loss do you use?
